src/Processors/Formats/Impl/ParallelFormattingOutputFormat.cpp

```diff
@@ -224,6 +224,7 @@ namespace DB
 
             /// Flush all the data to handmade buffer.
             formatter->flush();
+            out_buffer.finalize();
```

Comment:
That's maybe not enough. Some formats like `JSON` uses `WriteBufferValidUTF8` wrapper on top of this buffer and we should finalize if first. `formatter` have method `finalizeBuffers` for this purpose.

Reply:
Ah, we will flush all data from `WriteBufferValidUTF8 ` tou our buffer in `formatter->flush()`. I guess it's ok because we didn't remove `finalize` from `WriteBufferValidUTF8` destructor, so we won't get assert. But still better to call `finalizeBuffers` explicitely I think

Reply:
It is possible that It just works in some surprising ways. For example because some buffers do finalize in d-tor still.
I do not see the reason to remove all finalize calls from all buffers at once.

The rest buffers will be fixed in the following PRs.
Even more I will put my efforts into `is_finalize_required` in them. I really do not proud with all that lines with finalize calls in dummy buffers. But at least I hope that things are getting better.